### ISSUE TITLE
Sync subscriptions to file

### ISSUE THREAD

**Issue by zackhsi** [ViennaRSS/vienna-rss#818]
Hey, I have a hopefully simple feature ask: can I specify in preferences a file to sync my .opml subscriptions file to?

I want to back/versions up my subscriptions with Github via this preference.

Thanks!

**Comment by Eitot** [comment]
Do you mean a change in the default export location or some automatic export behaviour?

**Comment by zackhsi** [comment]
The latter. I would like Vienna to update a file every time the
subscriptions change. The file would always reflect the current state of
subscriptions.

**Comment by josh64x2** [comment]
@zackhsi a customised solution like you want is outside the scope of Vienna unfortunately. However you could take a look at the xml files in `~/Library/Application Support/Vienna/Sources` which contain a recent copy of each feeds xml.

**Comment by zackhsi** [comment]
I don't think it would be so much a customised solution, but rather a refactor. Instead of the export functionality parsing the list of feeds, generating an opml file, and writing it to the specified location... Vienna would always keep this opml file up to date, probably in `~/Library/Application Support/Vienna/Sources` as you mentioned, and then the "export' would really just be a copy of the file to the specified location.

We could put the hook to sync the opml at the end of this function: https://github.com/ViennaRSS/vienna-rss/blob/f438c43061bb4a92bc1766dc15b122e42333a9a6/src/AppController.m#L2699. It would really just call out to https://github.com/ViennaRSS/vienna-rss/blob/f438c43061bb4a92bc1766dc15b122e42333a9a6/src/Export.m#L83 and write to the default sync file location.

To summarize: instead of generating the opml file at export time, Vienna could generate it at subscribe time. And the generated file location could be customized :)

**Comment by josh64x2** [comment]
Oh ok sorry I misunderstood before.
It is probably a desirable feature from a backup point of view and could allow feeds to be safely stored in a mounted network or cloud volume.

**Comment by jlmendezbonini** [comment]
I didn't find any other active issues. Has there been any progress or other discussions about implementing this functionality?
